apiserver/facades/client/controller/controller.go

```diff
@@ -243,18 +266,11 @@ func (c *ControllerAPI) MongoVersion(ctx context.Context) (params.StringResult,
 // dashboard deployed on CAAS.
 func (c *ControllerAPI) dashboardConnectionInfoForCAAS(
 	ctx context.Context,
-	m *state.Model,
 	applicationName string,
 ) (*params.Proxy, error) {
-	configGetter := stateenvirons.EnvironConfigGetter{Model: m, CloudService: c.cloudService, CredentialService: c.credentialService}
-	environ, err := common.EnvironFuncForModel(m, c.cloudService, c.credentialService, configGetter)(ctx)
+	caasBroker, err := c.caasBrokerGetter(ctx)
 	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	caasBroker, ok := environ.(caas.Broker)
-	if !ok {
-		return nil, errors.New("cannot get CAAS environ for model")
+		return nil, fmt.Errorf("cannot get CAAS environ for model: %w", err)
```

Comment:
issue: error message does not match what work failed.

Reply:
I can't see the issue, what would be a better error message here?

Reply:
I misread.